enterprise/server/remote_execution/execution_server/execution_server.go

```diff
@@ -771,3 +775,35 @@ func (s *ExecutionServer) PublishOperation(stream repb.Execution_PublishOperatio
 		}
 	}
 }
+
+func (s *ExecutionServer) updateRouter(ctx context.Context, taskID string, executeResponse *repb.ExecuteResponse) error {
+	router := s.env.GetTaskRouter()
+	if router == nil {
+		return nil
+	}
+	// If the result was served from cache, nothing was actually executed, so no
+	// need to update the task router.
+	if executeResponse.GetCachedResult() {
+		return nil
+	}
+	// TODO: Maybe pre-fetch the command at the beginning of PublishOperation,
+	// since it requires 2 round trips to CAS.
+	instanceName, d, err := digest.ExtractDigestFromUploadResourceName(taskID)
+	if err != nil {
+		return status.WrapError(err, "parse upload resource name")
+	}
+	actionInstanceNameDigest := digest.NewInstanceNameDigest(d, instanceName)
+	action := &repb.Action{}
+	if err := cachetools.ReadProtoFromCAS(ctx, s.cache, actionInstanceNameDigest, action); err != nil {
+		return status.WrapError(err, "get action from CAS")
+	}
+	cmdDigest := action.GetCommandDigest()
+	cmdInstanceNameDigest := digest.NewInstanceNameDigest(cmdDigest, instanceName)
+	cmd := &repb.Command{}
+	if err := cachetools.ReadProtoFromCAS(ctx, s.cache, cmdInstanceNameDigest, cmd); err != nil {
+		return status.WrapError(err, "get command from CAS")
```

Comment:
please remove wrapError calls (and take a look at https://github.com/buildbuddy-io/buildbuddy/pull/534)

Reply:
Done